File: mesmaeker-java-mockito/src/main/java/fr/xgouchet/mesmaeker/mockito/OngoingReturningClause.java
package fr.xgouchet.mesmaeker.mockito;

import java.util.function.Consumer;
import java.util.function.Function;

/**
 * @author Xavier F. Gouchet
 */
public interface OngoingReturningClause<T, O> extends OngoingReverseClause<T> {


    OngoingReturningClause<T, O> when(Function<T, O> function);

}

File: mesmaeker-mockito/src/main/java/fr/xgouchet/mesmaeker/mockito/OngoingReturningClause.kt
package fr.xgouchet.mesmaeker.mockito

import java.util.function.Consumer
import java.util.function.Function

/**
 * @author Xavier F. Gouchet
 */
interface OngoingReturningClause<T, O> : OngoingReverseClause<T> {


    fun `when`(function: Function<T, O>): OngoingReturningClause<T, O>

    fun whenever(function: Function<T, O>): OngoingReturningClause<T, O>
}
